tests/main/ubuntu-core-classic/task.yaml

```diff
@@ -0,0 +1,40 @@
+summary: Ensure classic dimension works correctly
+systems: [ubuntu-core-16-64]
+environment:
+    SUDO_USER: test
+prepare: |
+    echo "test ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/create-test
+restore: |
+    rm -f /etc/sudoers.d/create-test
+execute: |
+    echo "Ensure classic can be installed"
+    snap install --devmode --beta classic
+    snap list|grep classic
+
+    echo "Check that classic can run commands inside classic"
+    classic test -f /var/lib/dpkg/status
+
+    echo "Ensure that after classic exits no processes are left behind"
+    classic "sleep 133713371337&"
+    if ps afx|grep 133713371337|grep -v grep; then
+        echo "The sleep process was not killed when classic exited"
+        echo "Something is wrong with the cleanup"
+        exit 1
+    fi
+
+    echo "Ensure sudo works without a password inside classic"
+    if [ "$(classic sudo id -u|tr -d "\r")" != "0" ]; then
```

Comment:
This works on a normal system but unfortunately classic (actually all snaps that need a pty) is slightly special because of bug  https://bugs.launchpad.net/snappy/+bug/1611493. I added a comment now:
```
+    # classic uses "script" to work around the issue that
+    # tty reports "no tty" inside snaps (LP: #1611493)
+    #
+    # "script" adds extra \r into the output that we need to filter here
```